### DISCUSSION TITLE
Calculer le salaire brut à partir du salaire net

### DISCUSSION THREAD

**Discussion by thom4parisot** [publicodes/publicodes#381] [Q&A] [answered]
Salut à vous,

Déjà, trop bien publicodes !
Encore mieux avec `modele-social`.

Je me serais attendu à obtenir salaire brut en faisant ceci : 

```js
import Engine, { formatValue } from 'publicodes'
import socialRules from 'modele-social'

const engine = new Engine(rules)

engine.setSituation({
  "salarié . contrat . date d'embauche": new Date('2023-01-01'),
  "salarié . contrat . statut cadre": 'non',
  "salarié . rémunération . net . à payer avant impôt": "2000 €/mois"
})
```

Et c'est le cas 👍🏻 

Par contre les éléments suivants (qui en découlent) sont marqués comme "pas encore défini" : 

- taux d'imposition (`engine.evaluate("impôt . taux d'imposition")`)
- net après prélèvement à la source(`engine.evaluate("salarié . rémunération . net . payé après impôt")`)
- coût total employeur (`engine.evaluate("salarié . coût total employeur")`)

Vous auriez une idée de la raison pour laquelle ces valeurs ne sont pas actualisées ?

Merci d'avance pour votre aide :-)

**Comment by johangirod** [comment, marked as answer]
Salut ! J'ai essayé et cela marche pour le net après impôt et le coût total employeur. 
https://codesandbox.io/s/focused-herschel-j6ywqz?file=/src/index.ts

Pour le taux d'imposition, il faut ajouter des éléments à la situation, tels qu'ils apparaissent dans la page de documentation : 
![image](https://github.com/betagouv/publicodes/assets/1775934/0faa373e-35b1-4d75-b3d7-5d485d57e603)

Pour plus d'info sur l'utilisation de la bibliothèque JS, voir la page de doc : https://mon-entreprise.urssaf.fr/d%C3%A9veloppeur/biblioth%C3%A8que-de-calcul

**Reply by thom4parisot** [reply]
Merci @johangirod ; en effet, ça coinçait parce que je rentrais d'autres paramètres dans le modèle.

[Je tentais de simuler](https://deploy-preview-8--solstice-www-next.netlify.app/simulateurs/) la situation d'une personne qui est entrepreneur·e-salarié·e en coopérative d'activité, donc ça mixait chiffre d'affaire et revenus.

Avec tes indications (spécifier des éléments de situation mentionées dans la capture d'écran), ça a généré les résultats attendus.

Merci 🙂
